lib/src/ctap2/requests/make_credential.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:cbor/cbor.dart';
+import '../constants.dart';
+import '../entities/credential_entities.dart';
+
+class MakeCredentialRequest {
+  final List<int> clientDataHash;
+  final PublicKeyCredentialRpEntity rp;
+  final PublicKeyCredentialUserEntity user;
+  final List<Map<String, dynamic>> pubKeyCredParams;
+  final List<PublicKeyCredentialDescriptor>? excludeList;
+  final Map<String, dynamic>? extensions;
+  final Map<String, bool>? options;
+  final List<int>? pinAuth;
+  final int? pinProtocol;
+  final bool? enterpriseAttestation;
+  
+  MakeCredentialRequest({
+    required this.clientDataHash,
+    required this.rp,
+    required this.user,
+    required this.pubKeyCredParams,
+    this.excludeList,
+    this.extensions,
+    this.options,
+    this.pinAuth,
+    this.pinProtocol,
+    this.enterpriseAttestation,
+  });
+}
+
+class MakeCredentialResponse {
+  final String fmt;
+  final List<int> authData;
+  final Map<String, dynamic> attStmt;
+  final bool? epAtt;
+  final List<int>? largeBlobKey;
+  
+  MakeCredentialResponse({
+    required this.fmt,
+    required this.authData,
+    required this.attStmt,
+    this.epAtt,
+    this.largeBlobKey,
+  });
+}
+
+class MakeCredentialUtils {
+  static List<int> makeMakeCredentialRequest(MakeCredentialRequest request) {
+    final map = <int, dynamic>{};
+    map[1] = CborBytes(request.clientDataHash);
```

Comment:
I suggest replacing these numbers with constants as well, for example, 1 could be named mcClientDataHashIdx